app/Repositories/JsonBaseRepository.php

```diff
@@ -33,5 +33,32 @@ public function create($item)
         return file_put_contents($this->filePath, json_encode($contents));
     }
 
+    public function allGroupedBy($field)
+    {
+        if (!in_array($field, $this->getEntityFields())) {
+            return throw new Exception("Field {$field} is not valid.");
+        }
+
+        return array_group($this->all(), $field);
+    }
+
+    public function isExist($condition = [])
+    {
+        foreach ($this->all() as $value) {
+            if (array_intersect_assoc($condition, $value) === $condition) {
```

Comment:
this logic can be moved to the separate function as it already used it two places